resolvectl: add page (#7484)
tldr-pages/tldr: pages/linux/resolvectl.md

# resolvectl

> Resolve domain names, IPV4 and IPv6 addresses, DNS resource records, and services.

> > Introspect and reconfigure the DNS resolver.

Reviewer: I don't think introspect is the correct word here.

Author: From  the man page:
"Resolve domain names, IPV4 and IPv6 addresses, DNS resource records, and services; introspect and reconfigure the DNS resolver."

Author: I believe it does support  [network introspection commands](https://www.ibm.com/docs/en/api-connect/5.0.x?topic=interface-network-introspection-commands), though I admit I didn't include introspection in the examples.

> More information: <https://www.freedesktop.org/software/systemd/man/resolvectl.html>.

- Show DNS settings:

`resolvectl status`

- Resolve the IPv4 and IPv6 addresses for one or more domains:

> `resolvectl query {{domain}}`

Reviewer 2: Are domains space separated? Perhaps we could indicate this somehow?

Reviewer 3: This way
```suggestion
`resolvectl query {{domain1 domain2 ...}}`
```

Author: Thanks @navarroaxel!

> - Retrieve the domain of {{IP}}:

Reviewer: ```suggestion
- Retrieve the domain of a specified IP:
```

> `resolvectl query {{ip}}`

Reviewer: ```suggestion
`resolvectl query {{IP}}`
```

> `resolvectl query {{IP}}`

Reviewer 3: We use `snake_case` for placeholders:
```suggestion
`resolvectl query {{ip_address}}`
```

> - Retrieve MX record of {{domain}}:

Reviewer: ```suggestion
- Retrieve an MX record of {{domain}}:
```

> - Retrieve an MX record of {{domain}}:

Reviewer 3: The token syntax here is not well rendered
```suggestion
- Retrieve an MX record of domain:
```

> `resolvectl  --legend=no -t MX query  {{domain}}`

Reviewer 3: We prefer long arguments, and there is extra spaces.
```suggestion
`resolvectl --legend={{no}} --type={{MX}} query {{domain}}`
```

Author: I selected commit suggestion, yet it still says 1 change requested. Is there anything else that needs to be changed?

- Resolve an SRV record, for example _xmpp-server._tcp gmail.com:

`resolvectl service _{{service}}._{{protocol}} {{name}}`

> - Retrieve the public key from an email address with an OPENPGP Record:

Reviewer: ```suggestion
- Retrieve the public key from an email address with a PGP Record:
```

Author: I made a typo here. The DNS record type is called OPENPGPKEY, probably the longest one available. What do you think about changing it to that?

Reviewer 2: What about this?
```suggestion
- Retrieve the public key from an email address from an OPENPGPKEY DNS record:
```
